src/test/java/com/cdancy/bitbucket/rest/features/UserApiMockTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.cdancy.bitbucket.rest.features;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Map;
+
+import org.testng.annotations.Test;
+
+import com.cdancy.bitbucket.rest.BaseBitbucketMockTest;
+import com.cdancy.bitbucket.rest.BitbucketApi;
+//import com.cdancy.bitbucket.rest.BitbucketApiMetadata;
+import com.cdancy.bitbucket.rest.domain.admin.UserPage;
+import com.google.common.collect.ImmutableMap;
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+
+@Test(groups = "unit", testName = "UserApiMockTest")
+public class UserApiMockTest extends BaseBitbucketMockTest {
+
+    private final String localGroup = "test";
+    private final String limitKeyword = "limit";
+    private final String startKeyword = "start";
+    private final String restApiPath = "/rest/api/";
+    private final String getMethod = "GET";
+
+
+    public void testUserListUsers() throws Exception {
+    
+    
+        final MockWebServer server = mockWebServer();
+
+        server.enqueue(new MockResponse()
+                .setBody(payloadFromResource("/user-list-users.json"))
+                .setResponseCode(200));
+        try (final BitbucketApi baseApi = api(server.getUrl("/"))) {
+
+            final UserPage up = baseApi.userApi().users("jcitizen", "stash-users", "ADMIN", null,
+                    null, null, null, null, null, 0, 2);
+            assertThat(up).isNotNull();
+            assertThat(up.errors()).isEmpty();
+            assertThat(up.size() == 1).isTrue();
+            assertThat(up.values().get(0).slug().equals("jcitizen")).isTrue();
+
+            final Map<String, ?> queryParams = ImmutableMap.of("filter", "jcitizen",
+                    "group","stash-users" , "permission", "ADMIN","start",0,"limit",2);
+            //assertSent(server, getMethod, restApiPath + BitbucketApiMetadata.API_VERSION
+            //        + "/users", queryParams);
```

Comment:
What is the failure you are seeing here?

Reply:
TravisCI is failing to pull, I want to show you the detailed errors from the build file. But I'm waiting for it to detect my commits.

Reply:
Do things fail locally when you run `./gradlew clean build mockTest`?

Reply:
Yes, they give me checkStyle error: Do not use Windows line endings

Reply:
Ok so not an issue with the test/assertion but more an issue with your gitconfig which you can play with to get this working as you'd expect.

Reply:
No, it doesn't even get to check assertions it fails from the beginning.

Reply:
Are you developing on windows or on a unix based machine? I do all development on unix so I'm not sure where that would be coming from. To fix things you _could_ just write the line out again :)

Reply:
I am using Ubuntu for my developments, I will try to figure out what's wrong and get back to you.

Reply:
I found out where the Mock errors were, also I removed that error from my checkStyle so it's easier to build locally, I didn't stage it tho.